src/main/java/org/jabref/gui/groups/GroupNodeViewModel.java

```diff
@@ -64,7 +66,9 @@ public GroupNodeViewModel(BibDatabaseContext databaseContext, StateManager state
         this.localDragBoard = Objects.requireNonNull(localDragBoard);
 
         LatexToUnicodeFormatter formatter = new LatexToUnicodeFormatter();
-        displayName = formatter.format(groupNode.getName());
+        displayNameProperty = new SimpleStringProperty(formatter.format(groupNode.getGroup().getName()));
```

Comment:
This should be
```suggestion
        displayNameProperty = new SimpleStringProperty();
        displayNameProperty.bind(Easybind.map(groupNode.getGroup().nameProperty(), formatter:format));
```
We want to update the displayed name every time the name of the group changes, using the formatter.